.github/workflows/xcodebuild-or-fastlane.yml

```diff
@@ -86,11 +86,6 @@ on:
         required: false
         type: boolean
         default: false
-      cacheDerivedData:
```

Comment:
What was the core idea to remove this flag here? We might want to ensure that we also keep https://github.com/StanfordBDHG/.github/blob/main/.github/workflows/xcodebuild-or-fastlane.yml in sync.

Reply:
Caching the derived data really isn't used and doesn't really seem to work reproducibly with Xcode in the CI setup.
However, we can also just keep the functionality. I thought I would just remove any unused functionality in one go.

Reply:
Sounds fair to me 👍